old issue back?

@n1zea144 we have a deployment where the exception mentioned here https://github.com/cBioPortal/cbioportal/pull/4611 : `Caused by: org.apache.ibatis.builder.IncompleteElementException: Could not find SQL statement to include with refid 'org.cbioportal.persistence.mybatis.PatientMapper.select'` 
shows up again when deploying v2.1.0. Any ideas what could be the cause?

@khzhu  I had those properties already set up. I just put moved `authenticate=saml` so it would be literally below `authorization=true`. After redeployment, I still get the same error. Here is the beginning of the stacktrace:

```
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'studyViewFilterApplier' defined in URL [jar:file:/usr/local/tomcat/webapps/cbioportal/WEB-INF/lib/web-2.1.0-SNAPSHOT.jar!/org/cbioportal/web/util/StudyViewFilterApplier.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'sampleServiceImpl': Unsatisfied dependency expressed through field 'studyService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'studyServiceImpl': Unsatisfied dependency expressed through field 'cancerTypeService'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cancerTypeServiceImpl': Invocation of init method failed; nested exception is org.apache.ibatis.builder.IncompleteElementException: Could not find SQL statement to include with refid 'org.cbioportal.persistence.mybatis.PatientMapper.select'
```

and further down here is the other piece of the stack trace

```
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'sampleServiceImpl': Unsatisfied dependency expressed through field 'studyService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'studyServiceImpl': Unsatisfied dependency expressed through field 'cancerTypeService'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cancerTypeServiceImpl': Invocation of init method failed; nested exception is org.apache.ibatis.builder.IncompleteElementException: Could not find SQL statement to include with refid 'org.cbioportal.persistence.mybatis.PatientMapper.select'
```

thanks @pieterlukasse @fedde-s !
For my case, the root cause more likely was the dbcp authentication failure (please see logs below). 
There was not any changes made to the database before updating cBioPortal from 2.0.1 to 2.1.0. Thus, user 'cbio_user' should not be denied, right? Only thing I could think of was the DataSource object was not initialized with user 'cbio_user' credential. 
In PR #4787 you proposed to replace a private static initDataSource() method with a new DataSource object creation. This newly instantiated **_dataSource_** object was pushed to the stack rather than stored in a heap space to make the connection available for future requests to the database.

> Caused by: org.mybatis.spring.MyBatisSystemException: nested exception is org.apache.ibatis.exceptions.PersistenceException:
Caused by: org.apache.ibatis.exceptions.PersistenceException:
Caused by: org.springframework.jdbc.CannotGetJdbcConnectionException: Could not get JDBC Connection; nested exception is java.sql.SQLException: Cannot create PoolableConnectionFactory (Access denied for user 'cbio_user'@'localhost' (using password: YES))
Caused by: java.sql.SQLException: Cannot create PoolableConnectionFactory (Access denied for user 'cbio_user'@'localhost' (using password: YES))
Caused by: java.sql.SQLException: Access denied for user 'cbio_user'@'localhost' (using password: YES)
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'sampleServiceImpl': Unsatisfied dependency expressed through field 'studyService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'studyServiceImpl': Unsatisfied dependency expressed through field 'cancerTypeService'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cancerTypeServiceImpl': Invocation of init method failed; nested exception is org.apache.ibatis.builder.IncompleteElementException: Could not find SQL statement to include with refid 'org.cbioportal.persistence.mybatis.PatientMapper.select'
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'studyServiceImpl': Unsatisfied dependency expressed through field 'cancerTypeService'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cancerTypeServiceImpl': Invocation of init method failed; nested exception is org.apache.ibatis.builder.IncompleteElementException: Could not find SQL statement to include with refid 'org.cbioportal.persistence.mybatis.PatientMapper.select'
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cancerTypeServiceImpl': Invocation of init method failed; nested exception is org.apache.ibatis.builder.IncompleteElementException: Could not find SQL statement to include with refid 'org.cbioportal.persistence.mybatis.PatientMapper.select'
Caused by: org.apache.ibatis.builder.IncompleteElementException: Could not find SQL statement to include with refid 'org.cbioportal.persistence.mybatis.PatientMapper.select'
Caused by: java.lang.IllegalArgumentException: XML fragments parsed from previous mappers does not contain value for org.cbioportal.persistence.mybatis.PatientMapper.select
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'sampleServiceImpl': Unsatisfied dependency expressed through field 'studyService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'studyServiceImpl': Unsatisfied dependency expressed through field 'cancerTypeService'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cancerTypeServiceImpl': Invocation of init method failed; nested exception is org.apache.ibatis.builder.IncompleteElementException: Could not find SQL statement to include with refid 'org.cbioportal.persistence.mybatis.PatientMapper.select'
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'studyServiceImpl': Unsatisfied dependency expressed through field 'cancerTypeService'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cancerTypeServiceImpl': Invocation of init method failed; nested exception is org.apache.ibatis.builder.IncompleteElementException: Could not find SQL statement to include with refid 'org.cbioportal.persistence.mybatis.PatientMapper.select'
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cancerTypeServiceImpl': Invocation of init method failed; nested exception is org.apache.ibatis.builder.IncompleteElementException: Could not find SQL statement to include with refid 'org.cbioportal.persistence.mybatis.PatientMapper.select'
Caused by: org.apache.ibatis.builder.IncompleteElementException: Could not find SQL statement to include with refid 'org.cbioportal.persistence.mybatis.PatientMapper.select'

@khzhu the error states:
```
Caused by: java.sql.SQLException: Access denied for user 'cbio_user'@'localhost' (using password: YES)
```
Is your DB running on `localhost`?

--- a/portal/src/main/webapp/WEB-INF/web.xml
+++ b/portal/src/main/webapp/WEB-INF/web.xml
@@ -32,7 +32,7 @@
 
     <context-param>
         <param-name>contextConfigLocation</param-name>
-        <param-value>classpath:applicationContext-security.xml classpath:applicationContext-business.xml classpath:applicationContext-core.xml classpath:applicationContext-logging.xml</param-value>
+        <param-value>classpath:applicationContext-business.xml classpath:applicationContext-security.xml classpath:applicationContext-core.xml classpath:applicationContext-logging.xml</param-value>
     </context-param>
 
     <context-param>

